src/App.js

```diff
@@ -1,43 +1,87 @@
 import React from "react";
 import Joke from "./components/Joke";
 import PageHeader from "./components/PageHeader";
+import SearchInput from "./components/SearchInput";
+import PropTypes from "prop-types";
 
 class App extends React.Component {
-  constructor() {
-    super()
-    this.state = {
-      jokeData: {
-        results: []
-      }
+  state = {
+    jokeData: {
+      results: []
+    },
+    searchTerm: "",
+    loadingState: "resting"
+  };
+
+  handleChange = event => {
+    this.setState({ searchTerm: event.target.value });
```

Comment:
You should use `currentTarget` here rather than `target`. And before you ask, [this](https://stackoverflow.com/questions/10086427/what-is-the-exact-difference-between-currenttarget-property-and-target-property) gives a good answer to the difference as well as a link out to a blog post explaining it.

Reply:
Thanks for adding a link here. Makes perfect sense in this case.